test/e2e/external_oidc_test.go

```diff
@@ -187,263 +276,388 @@ func newTestClient(t *testing.T) (*testClient, error) {
+		require.NoError(t, err)
+		certData = cm.Data["ca-bundle.crt"]
 	}
 
-	return
+	expectedAuthConfigJSON := fmt.Sprintf(`{"kind":"AuthenticationConfiguration","apiVersion":"apiserver.config.k8s.io/v1beta1","jwt":[{"issuer":{"url":"%s","certificateAuthority":"%s","audiences":[%s],"audienceMatchPolicy":"MatchAny"},"claimMappings":{"username":{"claim":"%s","prefix":"%s"},"groups":{"claim":"%s","prefix":"%s"},"uid":{}}}]}`,
+		idpURL,
+		strings.ReplaceAll(certData, "\n", "\\n"),
+		strings.Join([]string{fmt.Sprintf(`"%s"`, oidcClientId)}, ","),
+		authSpec.OIDCProviders[0].ClaimMappings.Username.Claim,
+		usernamePrefix,
+		groupsClaim,
+		groupsPrefix,
+	)
+
+	for _, cm := range []struct {
+		ns   string
+		name string
+	}{
+		{"openshift-config-managed", "auth-config"},
+		{"openshift-kube-apiserver", "auth-config"},
+		{"openshift-kube-apiserver", fmt.Sprintf("auth-config-%d", kasRevision)},
+	} {
+		actualCM, err := tc.kubeClient.CoreV1().ConfigMaps(cm.ns).Get(ctx, cm.name, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.Equal(t, expectedAuthConfigJSON, actualCM.Data["auth-config.json"], "unexpected auth-config.json contents in %s/%s", actualCM.Namespace, actualCM.Name)
+	}
 }
 
-func (tc *testClient) updateAuthForOIDC(ctx context.Context, idpURL, idpName string) (auth *configv1.Authentication, cleanups []func(), err error) {
-	tc.t.Log("will update auth CR for OIDC")
+type namespacedObject struct {
+	namespace string
+	name      string
+}
 
-	auth, err = tc.configClient.ConfigV1().Authentications().Get(ctx, "cluster", metav1.GetOptions{})
+func (tc *testClient) validateOAuthResources(ctx context.Context, requireMissing bool) (bool, error) {
+	dynamicClient, err := dynamic.NewForConfig(tc.kubeConfig)
 	if err != nil {
-		return
+		return false, fmt.Errorf("unexpected error while creating dynamic client: %v", err)
 	}
 
-	origSpec := auth.Spec.DeepCopy()
-	auth.Spec.Type = configv1.AuthenticationTypeOIDC
-	auth.Spec.WebhookTokenAuthenticator = nil
-	auth.Spec.OIDCProviders = []configv1.OIDCProvider{
-		{
-			Name: idpName,
-			Issuer: configv1.TokenIssuer{
-				URL:       idpURL,
-				Audiences: []configv1.TokenAudience{oidcAudience},
-				CertificateAuthority: configv1.ConfigMapNameReference{
-					Name: oidcCABundleConfigMap,
-				},
-			},
-			ClaimMappings: configv1.TokenClaimMappings{
-				Groups: configv1.PrefixedClaimMapping{
-					TokenClaimMapping: configv1.TokenClaimMapping{
-						Claim: oidcGroupsClaim,
-					},
-				},
-				Username: configv1.UsernameClaimMapping{
-					TokenClaimMapping: configv1.TokenClaimMapping{
-						Claim: oidcUsernameClaim,
-					},
-				},
-			},
-			OIDCClients: []configv1.OIDCClientConfig{
-				{
-					ClientID:           "console",
-					ClientSecret:       configv1.SecretNameReference{Name: "console-secret"},
-					ComponentName:      "console",
-					ComponentNamespace: "openshift-console",
-				},
-			},
-		},
+	for _, obj := range []struct {
+		gvr       string
```

Comment:
Any reason to not just make this `schema.GroupVersionResource`?

I know this makes the actual test definitions a bit more verbose, but it prevents an additional failure mode for the test of having to parse the string into the type.

Reply:
I opted for the full string representation as I feel it is a more familiar view of a fully qualified resource name, thus making it easier to read, at the expense of parsing it. And besides, since these are const strings, there wouldn't be any spontaneous failure of the test, maybe during development if we add something there.

Happy to change it to an object though :+1:

Reply:
While I think that is a reasonable argument, in a set of e2e tests that can already have numerous failure modes I'd much prefer to reduce those where possible. 

IMO the small sacrifice of readability for the gain of one less failure mode is a beneficial change for long running tests like this.